src/sql/workbench/browser/designer/tableActions.ts

```diff
@@ -107,6 +99,15 @@ export class MoveRowUpAction extends DesignerTableAction {
 			resolve();
 		});
 	}
+
+	public override updateState(row?: number): void {
+		if (row === 0 && this instanceof MoveRowUpAction) {
```

Comment:
why do we still need this instanceof MoveRowUpAction?

Reply:
because we are disabling that specific action based on the selected row index.

Reply:
you are already in the MoveRowUpAction class.

Reply:
sorry, this is what lack of sleep can do to you 😆